Lib/glyphsLib/classes.py

```diff
@@ -3227,46 +3283,47 @@ def exports(self):
     def exports(self, value):
         self.active = value
 
-    def _get_from_properties(self, key):
-        for p in self.properties:
-            if p.key == key:
-                return p.defaultValue
-        return None
-
     @property
     def familyName(self):
-        value = (
-            self._get_from_properties("familyNames")
+        return (
+            self.properties.get("familyNames")
             or self.customParameters["familyName"]
+            or self.parent.familyName
         )
-        if value:
-            return value
-        return self.parent.familyName
 
     @familyName.setter
     def familyName(self, value):
+        # TODO: Update this to write into either custom parameters or properties
+        # depending on the version.
         self.customParameters["familyName"] = value
 
     @property
     def preferredFamily(self):
-        value = self.customParameters["preferredFamily"]
-        if value:
-            return value
-        return self.parent.familyName
+        return self.customParameters["preferredFamily"] or self.parent.familyName
 
     @preferredFamily.setter
     def preferredFamily(self, value):
         self.customParameters["preferredFamily"] = value
 
+    @property
+    def preferredFamilyName(self):
+        return (
+            self.properties.get("preferredFamilyNames")
+            or self.customParameters["preferredFamilyName"]
+        )
+
     @property
     def preferredSubfamilyName(self):
-        value = self.customParameters["preferredSubfamilyName"]
-        if value:
-            return value
-        return self.name
+        return (
+            self.properties.get("preferredSubfamilyNames")
+            or self.customParameters["preferredSubfamilyName"]
+            or self.name
```

Comment:
same here actually...

```python
print(Glyphs.font.instances[0].name)
print(Glyphs.font.instances[0].preferredSubfamilyName)
```

gives me

```
Regular
None
```

Reply:
Changed. Uncomfortable that no test had to be updated.

Reply:
yeah.. let's say these accessors are provided at best-effort basis.. You're welcome to add tests of course

Reply:
Ah, already merged. But I will add tests for this in a separate PR still!